syncd/FlexCounter.cpp

```diff
@@ -2112,6 +2214,19 @@ void FlexCounter::addCounter(
 
             setSwitchDebugCounterList(vid, rid, switchCounterIds);
         }
+        else if (objectType == SAI_OBJECT_TYPE_MACSEC_SA && field == MACSEC_SA_ATTR_ID_LIST)
+        {
+            std::vector<sai_macsec_sa_attr_t> MACsecSAIds;
+
+            for (const auto &str : idStrings)
+            {
+                sai_macsec_sa_attr_t attr;
+                sai_deserialize_macsec_sa_attr(str.c_str(), attr);
```

Comment:
use direct enum deserialize here, you will not be using this function anywhere else